src/script/api/script_airport.cpp

```diff
@@ -170,3 +170,28 @@
 
 	return (int64_t)GetMaintenanceCostFactor(type) * _price[PR_INFRASTRUCTURE_AIRPORT] >> 3;
 }
+
+/* static */ SQInteger ScriptAirport::CanPlaneUseAirport(AirportType airport_type, PlaneType plane_type)
+{
+	if (!IsAirportInformationAvailable(airport_type)) return -1;
+	if (plane_type != PT_HELICOPTER && plane_type != PT_SMALL_PLANE && plane_type != PT_BIG_PLANE) return -1;
```

Comment:
There's no reason to check the `plane_type` before the switch, IMHO. Early returns are useful to avoid unnecessary computing or an unsafe check, but neither of those happen here and this just adds complexity. You need to deal with the "none of the above" case in the switch already, so do it all there instead of splitting it.
```suggestion

```